src/main/java/com/movie/store/controllers/CinemaHallController.java

```diff
@@ -0,0 +1,40 @@
+package com.movie.store.controllers;
+
+import com.movie.store.model.CinemaHall;
+import com.movie.store.model.dto.CinemaHallRequestDto;
+import com.movie.store.model.dto.CinemaHallResponseDto;
+import com.movie.store.service.interfaces.CinemaHallService;
+import com.movie.store.service.mappers.CinemaHallMapper;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/cinema-halls")
+public class CinemaHallController {
+    private CinemaHallService cinemaHallService;
+    private CinemaHallMapper cinemaHallMapper;
+
+    @Autowired
+    public CinemaHallController(CinemaHallService cinemaHallService,
+                                CinemaHallMapper cinemaHallMapper) {
+        this.cinemaHallService = cinemaHallService;
+        this.cinemaHallMapper = cinemaHallMapper;
+    }
+
+    @PostMapping
+    public void addCinemaHalls(@RequestBody CinemaHallRequestDto cinemaHallRequestDto) {
+        CinemaHall cinemaHall = cinemaHallMapper.mapRequestDtoToCinemaHall(cinemaHallRequestDto);
+        cinemaHallService.add(cinemaHall);
+    }
+
+    @GetMapping
+    public List<CinemaHallResponseDto> getAllCinemaHall() {
+        return cinemaHallMapper.mapAllCinemaHallToResponseDto(
+                cinemaHallService.getAll());
```

Comment:
what if you will need to map only one cinema hall? let's make the method in mapper take only one object

and call the method for each cinema hall